src/classrooms/classrooms.service.ts

```diff
@@ -51,20 +52,34 @@ export class ClassroomsService {
   }
 
   async getClassroomsForUser(userId: number): Promise<Classroom[]> {
-    const user = await this.em.findOne(User, { id: userId }, { populate: ["teacher"] });
-    if (!user) {
-      throw new NotFoundException("User not found");
-    }
-
-    const classrooms = await this.em.find(
-      Classroom,
-      { teacher: user.teacher },
-      {
-        orderBy: { classTime: "DESC" },
-        populate: ["teacher"],
-      },
+    const user = await this.userRepository.findOneOrFail(
+      { id: userId },
+      { populate: ["teacher", "student"] },
     );
-    // TODO: query for fetching classrooms that a student is enrolled in.
+    let classrooms: Classroom[] = [];
+    if (user.student) {
+      classrooms = await this.classroomsRepository.find(
+        {
+          id: {
+            $in: (
+              await this.em.find(Enrollment, { student: user.student })
+            ).map((enrollment) => enrollment.classroom.id),
+          },
+        },
+        {
+          populate: ["teacher", "teacher.user"],
+          orderBy: { classTime: "DESC" },
+        },
+      );
```

Comment:
Let's make this a repository method with an appropriate name. 
Also, I think there is scope to make more readable. There should not be a query inside a clause of another query.